src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.IncrementalAnalyzerProcessor.cs

```diff
@@ -182,30 +190,29 @@ private void ResetLogAggregator()
                     Func<IIncrementalAnalyzer, T, CancellationToken, Task> runnerAsync,
                     CancellationToken cancellationToken)
                 {
-                    // this is a best effort progress report. since we don't clear up
-                    // reported progress when work is done or cancelled. it is possible
-                    // that last reported work is left in report even if it is already processed.
-                    // but when everything is finished, report should get cleared
-                    _ = _registration.ProgressReporter.Update(GetFilePath(value));
-
-                    foreach (var analyzer in analyzers)
+                    using (_registration.ProgressReporter.Evaluating())
```

Comment:
💡 I believe a using declaration here will simplify the code, and definitely simplify the diff

```suggestion
                    using var evaluating = _registration.ProgressReporter.Evaluating();
```